crates/rpc/rpc-types/src/eth/fee.rs

```diff
@@ -20,3 +23,29 @@ pub struct FeeHistory {
     #[serde(default)]
     pub reward: Option<Vec<Vec<U256>>>,
 }
+
+/// LRU cache for `eth_feeHistory` RPC method. Block Number => Fee History.
+#[derive(Clone, Debug)]
+pub struct FeeHistoryCache(pub Arc<Mutex<LruCache<BlockNumber, FeeHistoryCacheItem>>>);
```

Comment:
since the mutex needs to lock for db seeks, I think we should use a tokio Mutex instead, so other concurrent requests are non-blocking

Reply:
fixed